src/Orleans/Serialization/NonSerializableAttribute.cs

```diff
@@ -0,0 +1,12 @@
+﻿namespace Orleans.Serialization
+{
+    using System;
+
+    /// <summary>
+    /// Marker attribute for classes which should not be serialized.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class)]
+    public class NonSerializableAttribute : Attribute
```

Comment:
Yes, unfortunately `[NonSerialized]` is not available for classes. I only use this for one class, `SiloAssemblyLoader`, which easily could be serializable except that the default constructor does work (scanning for assemblies), so I think it shouldn't. Happy to remove if you prefer.